### ISSUE TITLE
Lists of objects can't be parsed

### ISSUE THREAD

**Issue by jpd236** [Him188/yamlkt#1]
Trying to parse some YAML from a multiplatform project and thought I'd give this project a try - thank you for writing it :)

When attempting to serialize a list of a nested class, the deserializer fails because it does not expect the "-" token while parsing a class object. It seems like if it sees a "-" while parsing a list item, it should finish that object and move onto a new one rather than aborting.

Here's a test case to validate any fixes:

```kotlin
    @Test
    fun testNestedList() {
        @Serializable
        data class NestedTestData(val item1: String, val item2: String)

        @Serializable
        data class TestData(val list: List<List<NestedTestData>>)

        assertEquals(
            TestData(
                listOf(
                    listOf(
                        NestedTestData("1", "A"),
                        NestedTestData("2", "B")
                    ),
                    listOf(
                        NestedTestData("3", "C")
                    )
                )
            ),
            default.parse(
                TestData.serializer(), """
                    list:
                    - - item1: 1
                        item2: A
                      - item1: 2
                        item2: B
                    - - item1: 3
                        item2: C
                """.trimIndent()
            )
        )
    }
```

**Comment by Him188** [comment]
Thank you for reporting! I will work on this soon.

**Comment by Him188** [comment]
As a workaround (by not changing your current logic and data structure), use FLOW_MAP and FLOW_SEQUENCE in YamlConfiguration. These two are more stable now.

```kotlin
val yaml = Yaml(classSerialization = YamlConfiguration.FLOW_MAP)
```

**Comment by jpd236** [comment]
Thanks for the quick fix and workaround!

**Comment by jpd236** [comment]
FYI, with the above unit test, changing `default` to:

```kotlin
Yaml(YamlConfiguration(
        listSerialization = YamlConfiguration.ListSerialization.FLOW_SEQUENCE, 
        mapSerialization = YamlConfiguration.MapSerialization.FLOW_MAP))
```

still fails with the 0.2.0 release. The fix you checked in is working for me though.

**Comment by Him188** [comment]
@jpd236 
I just released `0.3.0`. Thank you for your help
